### DISCUSSION TITLE
[Suggestion] Solar Panels

### DISCUSSION THREAD

**Discussion by Regnion** [tsunamayo/Starship-EVO#4005] [Suggestions]
While a great big main reactor is always cool to have, it wouldn't hurt to have an alternative option for power generation. Consider remote outposts or small, unmanned layover stations. Some panels would serve for keeping the lights on at least. Elsewhere, they can act as a good backup / emergency power source in the event the reactors run out of fuel or are damaged.

Fuel can feed a reactor to generate power, and fuel can feed thrusters to generate thrust- but having energy produced by panels instead means you can cut the first part out. Electric propulsion also means you can cut fuel entirely- but your thruster performance will be significantly less. So a solar sailer would be sorely slower, but solely solar.

Perhaps panels get more effective the closer they are to the sun- and more panels also means larger surface area to regulate heat. This won't mean you'll be dipping into the surface of the sun, but you could potentially get closer than many other "standard" ships, specifically to harvest off energy from its plasmic radiance. Solar panels must require exposure to sunlight. They should be constrained to 1 block and/or 1 brick in thickness, but stretchable in length and width. Stacking them would not work.

The idea here is solar skippers would move energy product, much like fuel tankers move fuel. Specially designed star-orbiting power stations would be somewhat dangerous for normal ships to visit- but specially designed vessels with large wings, and/or solar panel sails, would be able to access them and bring the energy back in batteries to safer space for market.

For inspiration, I provide only a large, rectangular solar panel array suitable for a large space station.

![20210319224143_1](https://user-images.githubusercontent.com/13087952/111860925-382eb800-8910-11eb-9c27-4445ac04384f.jpg)

Praise the Sun,

Regnion

**Comment by AlienXtream** [comment]
made a similar suggestion on one of the "dev asks" threads. i also suggested that they provide "low" power. basically only powering systems like item/fuel pipes, lights, logic and doors but not higher need systems like shields or non fueled engines. no matter how many you have they will only ever power those systems. 

>"Solar panels must require exposure to sunlight. They should be constrained to 1 block and/or 1 brick in thickness, but stretchable in length and width. Stacking them would not work."

my idea was to keep them even simpler than that and not really care about line of sight so long as they are not blocked (i.e no blocks in the immediate area in front of the panel. think big box that is invisible but must have no voxels. trying to think minimal performance cost here and ray casts can be slow)

i also suggested "solar paints" (which is a real, albeit mostly proof of concept stage, technology) and variable stars (some would make panels work better while others would produce less power as well as other effects not power related).

**Comment by TIKIRobo** [comment]
it would be nice if the solar panel blocks were in the normal shapes so that we can make sloped panels or curved ones without having to mess around with child entities too much

**Comment by Regnion** [comment]
> 
> 
> made a similar suggestion on one of the "dev asks" threads. i also suggested that they provide "low" power. basically only powering systems like item/fuel pipes, lights, logic and doors but not higher need systems like shields or non fueled engines. no matter how many you have they will only ever power those systems.
> 
> > "Solar panels must require exposure to sunlight. They should be constrained to 1 block and/or 1 brick in thickness, but stretchable in length and width. Stacking them would not work."
> 
> my idea was to keep them even simpler than that and not really care about line of sight so long as they are not blocked (i.e no blocks in the immediate area in front of the panel. think big box that is invisible but must have no voxels. trying to think minimal performance cost here and ray casts can be slow)
> 
> i also suggested "solar paints" (which is a real, albeit mostly proof of concept stage, technology) and variable stars (some would make panels work better while others would produce less power as well as other effects not power related).

I saw your suggestion there, yes, but his topic wasn't really about this system specifically, it was moreso about station modules. He also never acknowledged the suggestion, so he might not have seen it. This post is to be of singular focus, and specific.

I wouldn't recommend limiting solar panels to subsystems. I would rather have a new, electric engine, and batteries, which could then power shields and electric thrusters. I do think standard thrusters should be fuel based.
